### PR TITLE
github: modify codecov.yml

### PR DESCRIPTION
**Pull request by arvindbr8**
A few improvements to codecov settings

+ Change checks to be non-blocking: https://docs.codecov.com/docs/common-recipe-list#set-non-blocking-status-checks
+ Ignore paths explicitly [avoiding globs as much as possible]
+ add `diff` and `files` to PR comment: https://docs.codecov.com/docs/pull-request-comments#layout

RELEASE NOTES: n/a

### CODE REVIEW

File: .github/codecov.yml [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
+        informational: true
+    patch:
+      default:
+        informational: true
 ignore:
-  - "test/"
+  - "*test*"
   - "benchmark/"
-  - "internal/testutils"
+  - "*testutils*"
```
Comment (easwars) [inline .github/codecov.yml:12]: This will be covered by `*test*` right?

Comment (easwars) [reply .github/codecov.yml:12]: Also, I'm a little concerned about using wildcards here. How many entries do you have if you were to not use wildcards and instead listed them out explicitly?

Comment (arvindbr8) [reply .github/codecov.yml:12]: I actually agree with you. I'll make it explicit since its not that big of a list

File: .github/codecov.yml [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
+    patch:
+      default:
+        informational: true
 ignore:
-  - "test/"
-  - "benchmark/"
-  - "internal/testutils"
-  - ".*.pb.go"
+  - "benchmark/**/*"
+  - "stress/**/*"
```
Comment (dfawley) [inline .github/codecov.yml:11]: This was removed.

File: .github/codecov.yml [hunk 3]
```diff
@@ -1,16 +1,27 @@
...
+  # All 'pb.go's
+  - "**/*.pb.go"
+  # Tests and test related files.
+  - "test/**/*"
```
Comment (dfawley) [inline .github/codecov.yml:15]: `**/test` instead of anything with test?

Similarly:
- `**/testutils`
- `**/testdata`

File: .github/codecov.yml [hunk 4]
```diff
@@ -1,16 +1,27 @@
...
+  - "xds/internal/testutils/**/*"
+  - "xds/internal/test/**/*"
+  - "balancer/rls/internal/test/e2e/**/*"
+  - "testdata"
+  # Other submodules.
+  - "security/**/*"
```
Comment (dfawley) [inline .github/codecov.yml:21]: "security"?

File: .github/codecov.yml (hunk 2 above)
Comment (arvindbr8) [reply .github/codecov.yml]: Done

File: .github/codecov.yml (hunk 3 above)
Comment (arvindbr8) [reply .github/codecov.yml]: Done

File: .github/codecov.yml (hunk 4 above)
Comment (arvindbr8) [reply .github/codecov.yml]: Done

**Review by dfawley — APPROVED** [review]

**Pull request merged**